compiler/rustc_borrowck/src/universal_regions.rs

```diff
@@ -444,12 +450,9 @@ impl<'cx, 'tcx> UniversalRegionsBuilder<'cx, 'tcx> {
             bound_inputs_and_output,
             &mut indices,
         );
-        // Converse of above, if this is a function then the late-bound regions declared on its
-        // signature are local to the fn.
-        if self.mir_def.did.to_def_id() == typeck_root_def_id {
-            self.infcx
-                .replace_late_bound_regions_with_nll_infer_vars(self.mir_def.did, &mut indices);
-        }
+        // Converse of above, if this is a function/closure then the late-bound regions declared on its
+        // signature are local.
+        self.infcx.replace_late_bound_regions_with_nll_infer_vars(self.mir_def.did, &mut indices);
```

Comment:
This may register free regions that come from outer bodies as "local". Is this correct?
The comment probably needs to be updated.

Reply:
It doesn't, but only because we call `replace_late_bound_regions_with_nll_infer_vars` once before, but with the parent def-id, and we filter out infer vars that have already been added. This is somewhat confusing behavior -- I will change this call to be clearer.

Reply:
Replaced with a check on the def_kind for is_fn_like

Reply:
I made this a bit clearer in a subsequent commit